### ISSUE TITLE
Detect fmp4 support

### ISSUE THREAD

**Issue by jfizz** [videojs/videojs-contrib-hls#1187]
Is there a way to determine if a browser supports fmp4? I have tried checking the existence of window.MediaSource but that check fails on various flavors of IE/Edge/Android.

**Comment by mjneil** [comment]
Any browser that supports MSE would be able to support playing back fmp4 when using a library like this one. If you mean what browsers support an fmp4 HLS manifest natively without this project, then only browsers that support MPEG2TS HLS natively (e.g. Edge, Safari) should be able to play back the manifest.

**Comment by jfizz** [comment]
Sorry about that, I wasn't clear. I am using the latest version of videojs-contrib-hls to play HLS/fmp4 streams. I recently did some browser testing and was not able to get fmp4 streams to work on a few browsers on Browserstack. Also note that the default video (non fragmented mp4) worked fine on all the browsers below.

Player: http://videojs.github.io/videojs-contrib-hls/
Playlist: https://d2zihajmogu5jn.cloudfront.net/advanced-fmp4/master.m3u8

- IE 11 
Actually works. I was using the wrong version of windows when I tested initially.

- Win10 - Edge 14
No error, but showed the loader for a minute before I gave up. I can confirm that the playlist file was downloaded successfully by checking out the network panel.
 
- Win10 - Edge 15 (Latest)
`{
  code: 4,
  MEDIA_ERR_ABORTED: 1,
  MEDIA_ERR_CUSTOM: 0,
  MEDIA_ERR_DECODE: 3,
  MEDIA_ERR_ENCRYPTED: 5,
  MEDIA_ERR_NETWORK: 2,
  MEDIA_ERR_SRC_NOT_SUPPORTED: 4,
  message: "The media could not be loaded, either because the server or network failed or because the format is not supported.",
  status: null
}`

- Galaxy S7 - Chrome 58 (!!MediaSource === true)
`{
  code: 3,
  message: "The media playback was aborted due to a corruption problem or because the media used features your browser did not support."
}`
